controllers/events.js

```diff
@@ -173,22 +182,86 @@ const updateEvent = async (req, res) => {
  * @param {Object} req - The Express request object.
  * @param {Object} res - The Express response object.
  * @returns {Promise<Object>} The JSON response with a message indicating the session has ended.
- * @throws {Object} The JSON response with an error message if an error occurred while ending the event.
+ * @returns {Promise<Object>} The JSON response with an error message if an error occurred while ending the event.
  */
 const endActiveEvent = async (req, res) => {
```

Comment:
endActiveEvent() function is not checking the lock property of the req.body object. This means that the function will allow users to end active events, even if the event is locked.

 to fix this change to below
if (payload.lock === true) {
  return res.status(403).json({
    error: 'Forbidden',
    message: 'The event is locked and cannot be ended',
  });
}